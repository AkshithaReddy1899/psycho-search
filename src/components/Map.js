import React from 'react';
import {
  ComposableMap,
  Geographies,
  Geography,
  Marker,
  Sphere,
} from 'react-simple-maps';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const geoUrl = 'https://raw.githubusercontent.com/zcreativelabs/react-simple-maps/master/topojson-maps/world-110m.json';

function Map(props) {
  const { markers } = props;

  return (
    <div className="container-lg" style={{ width: '950px', height: '650px' }}>
      <ComposableMap projection="geoMercator" width={950} height={650} projectionConfig={{ rotate: [-10, 0, 0], scale: 147 }}>

        <Sphere stroke="#000" />
        <Geographies geography={geoUrl}>
          {({ geographies }) => geographies.map((geo) => (
            <Geography
              key={geo.rsmKey}
              geography={geo}
              fill="#EAEAEC"
              stroke="#D6D6DA"
            />
          ))}
        </Geographies>
        {markers.map(({ id, coordinates }) => (
          <Link to='/person' key={id} state= {{id: id}}>
            <Marker key={id} coordinates={coordinates}>
            <g
              fill="none"
              stroke="#FF5533"
              strokeWidth="3"
              strokeLinecap="round"
              strokeLinejoin="round"
            >
              <circle cx="12" cy="10" r="2" />
              <path d="M12 21.7C17.3 17 20 13 20 10a8 8 0 1 0-16 0c0 3 2.7 6.9 8 11.7z" />
            </g>
            </Marker>
          </Link>
        ))}
      </ComposableMap>
    </div>
  );
}

Map.defaultProps = {
  markers: [
    {
      coordinates: [],
      name: 'Name',
      id: 0,
    },
  ],
};

Map.propTypes = {
  markers: PropTypes.arrayOf(PropTypes.shape({
    coordinates: PropTypes.arrayOf,
    name: PropTypes.string,
    id: PropTypes.number,
  })),
};

export default Map;
