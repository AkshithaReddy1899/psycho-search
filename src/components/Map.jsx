/* eslint-disable func-names */
/* eslint-disable react/jsx-no-bind */
/* eslint-disable new-cap */
import React from 'react';
import {
  MapContainer, TileLayer, Marker, Popup,
} from 'react-leaflet';
import { useNavigate } from 'react-router-dom';
import MarkerClusterGroup from 'react-leaflet-cluster';
import { Icon, divIcon, point } from 'leaflet';
import { data } from '../assets/data';

function MapComponent() {
  const navigate = useNavigate();
  const customIcon = new Icon({
    iconUrl: 'https://cdn-icons-png.flaticon.com/512/447/447031.png',
    iconSize: [38, 38],
  });

  // custom cluster icon
  const createClusterCustomIcon = function (cluster) {
    return new divIcon({
      html: `<span class="cluster-icon">${cluster.getChildCount()}</span>`,
      className: 'custom-marker-cluster',
      iconSize: point(33, 33, true),
    });
  };

  return (
    <center>
      <div id="map" className="mx-2">
        <MapContainer className="leaflet-container" center={data[0].geocode} zoom={13} scrollWheelZoom={false}>
          <TileLayer
            className="tile-layer-class"
            attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
          />
          <MarkerClusterGroup
            chunkedLoading
            iconCreateFunction={createClusterCustomIcon}
          >
            {data.map((marker) => (
              <Marker
                key={marker.id}
                position={marker.geocode}
                icon={customIcon}
                onClick={navigate}
                eventHandlers={{
                  click: () => {
                    navigate('/person', { state: { id: marker.id } });
                  },
                }}
              >
                <Popup>{marker.name}</Popup>
              </Marker>
            ))}
          </MarkerClusterGroup>
        </MapContainer>
      </div>
    </center>
  );
}

export default MapComponent;
