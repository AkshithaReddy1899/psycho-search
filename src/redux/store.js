import {
  legacy_createStore as createStore,
  combineReducers,
  applyMiddleware,
  compose,
} from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import personsReducer from './person';

const rootReducer = combineReducers({
  person: personsReducer,
});

const store = createStore(rootReducer, compose(applyMiddleware(logger, thunk)));

export default store;
