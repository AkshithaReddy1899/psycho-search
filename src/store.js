import { configureStore } from '@reduxjs/toolkit';
import personSlice from './slices/personSlice';

const reducer = {
  person: personSlice,
};

const store = configureStore({
  reducer,
  devTools: true,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: false,
  }),
});

export default store;
