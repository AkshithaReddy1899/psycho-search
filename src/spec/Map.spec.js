import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import matches from './test-utils';
import store from '../redux/store';
import Map from '../components/Map';

describe('Home', () => {
  it('Home matched snapshot', () => {
    matches(<Provider store={store}><BrowserRouter><Map /></BrowserRouter></Provider>);
  });
});
