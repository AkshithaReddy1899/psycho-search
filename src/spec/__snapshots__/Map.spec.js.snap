// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home matched snapshot 1`] = `
<div
  className="container-lg"
>
  <svg
    className="rsm-svg "
    data-tip=""
    viewBox="0 0 950 650"
  >
    <g
      className="rsm-geographies "
    />
    <a
      href="/person"
      onClick={[Function]}
    >
      <g
        className="rsm-marker "
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        transform="translate(NaN, NaN)"
      >
        <g
          fill="none"
          stroke="#FF5533"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="3"
        >
          <circle
            cx="12"
            cy="10"
            r="2"
          />
          <path
            d="M12 21.7C17.3 17 20 13 20 10a8 8 0 1 0-16 0c0 3 2.7 6.9 8 11.7z"
          />
        </g>
      </g>
    </a>
  </svg>
</div>
`;
